docs/finality-provider-operation.md

```diff
@@ -939,4 +943,43 @@ For Finality Provider:
 
 > 🔒 **Security Note**: While database files can be recreated, loss of private keys in the keyring directories is **irrecoverable** and will result in permanent loss of your finality provider position and accumulated rewards.
 
-> 💡 **Future Updates**: The ability to recreate the `fpd.db` from chain state will be offered in future releases, but keyring backup remains critical.
+### 7.3 Recover finality-provider.db
+
+The `finality-provider.db` file contains both the finality provider's running
+status and the public randomness merkle proof. Either information loss
+compromised will lead to service halt, but they are recoverable.
+
+#### 7.3.1 Recover local status of a finality provider
+
+The local status of a finality provider is defined as follows:
+
+```go
+type StoredFinalityProvider struct {
+	FPAddr          string
+	BtcPk           *btcec.PublicKey
+	Description     *stakingtypes.Description
+	Commission      *sdkmath.LegacyDec
+	ChainID         string
+	LastVotedHeight uint64
+	Status          proto.FinalityProviderStatus
+}
+```
+
+It can be recovered by downloading the finality provider's info from the
+Babylon chain. Specifically, this can be achieved by repeating the
+[creation process](#61-create-finality-provider). The `create-finality-provider`
+cmd will download the info of the finality provider locally if it is already
+registered on Babylon.
+
+#### 7.3.2 Recover public randomness proof
+
+Every finality vote must contain the public randomness proof to prove that the
+randomness used in the signature is already committed on Babylon. Loss of
+public randomness proof leads to direct failure of the vote submission.
```

Comment:
We should add here at some point that the fpd must be stoped to run this command, otherwise it keeps the db locked

Reply:
Good point!